### DISCUSSION TITLE
Chinese display garbled code

### DISCUSSION THREAD

**Discussion by TianBoxv587** [Taitava/obsidian-shellcommands#179] [Debugging and testing]
![image](https://user-images.githubusercontent.com/91042118/157866165-eac701c4-a317-4677-aa83-f5acb5726453.png)

**Comment by Taitava** [comment]
Hi, thank you for your message! 🙂 

1. What version of the _Shell commands_ plugin do you have? `0.11.1`?
2. Judging from the `` ` `` escape characters in the shell command preview text, it seems you are using Windows and PowerShell. Can you try prefixing your command with the following:

``` PowerShell
$OutputEncoding = [console]::InputEncoding = [console]::OutputEncoding = New-Object System.Text.UTF8Encoding;
```

So the full command will be:

``` PowerShell
$OutputEncoding = [console]::InputEncoding = [console]::OutputEncoding = New-Object System.Text.UTF8Encoding; ECHO {{file_path:absolute}}
```

It should set certain encoding settings to `utf-8`, which the _Shell commands_ plugin can better understand.

Does it help? It's not a clean fix, but rather a quick and temporary workaround. I'm still thinking how to fix this properly. More about it here: #157

**Reply by Taitava** [reply]
Hi @TianBoxv587 ! 🙂 Do you still have this problem?
